README.md

```diff
@@ -76,317 +110,607 @@ cartridge --help
 
 These commands are supported:
 
-* `create` - create a new app from template;
-* `pack` - pack application into a distributable bundle;
-* `build` - build application for local development;
+* `create` - create a new application from template;
+* `build` - build the application for local development;
 * `start` - start a Tarantool instance(s);
 * `stop` - stop a Tarantool instance(s).
+* `pack` - pack the application into a distributable bundle;
+
+### An application's lifecycle
+
+In a nutshell:
+
+1. [Create](#creating-an-application-from-template) an application
+   (e.g. `myapp`) from template:
+
+   ```sh
+   cartridge create --name myapp
+   ```
+
+2. [Build](#building-an-application) the application for local testing
+   (e.g. in `./myapp`):
+
+   ```sh
+   cartridge build ./myapp
+   ```
+
+3. [Run](#startingstopping-an-application-locally) instances locally:
 
-## Applications lifecycle
+   ```sh
+   cartridge start
+   cartridge stop
+   ```
 
-Create an application from a template:
+4. [Pack](#packing-an-application) the application into a distributable
+   (e.g. into an RPM package):
+
+   ```sh
+   cartridge pack rpm ./myapp
+   ```
+
+### Creating an application from template
+
+To create an application from the Cartridge template, say this in any directory:
 
 ```sh
-cartridge create --name myapp
+cartridge create --name <app_name> /path/to/
 ```
 
-Build an application:
+This will:
+
+* automatically set up a Git repository in a new `/path/to/<app_name>/`
+  directory,
+* tag it with [version](https://www.tarantool.io/en/doc/latest/book/cartridge/cartridge_dev/#application-versioning) `0.1.0`,
+* and put all necessary files into it.
+
+In this Git repository, you can:
+
+* develop your application by simply editing the default files provided
+  by the template,
+* plug all necessary modules,
+* and easily pack everything to deploy on your server(s).
+
+The template creates the `<app_name>/` directory with the following
+contents inside:
+
+* `<app_name>-scm-1.rockspec` file where you can specify application
+  dependencies.
+* ``deps.sh`` script that resolves the dependencies from the `.rockspec` file.
+* ``init.lua`` file which is the entry point for your application.
+* ``.git`` file necessary for a Git repository.
+* ``.gitignore`` file where you can specify the files to ignore.
+* ``env.lua`` file that sets common rock paths so that the application can be
+  started from any directory.
+* ``custom-role.lua`` file that is a placeholder for a custom (user-defined)
+  [cluster role](https://www.tarantool.io/en/doc/latest/book/cartridge/cartridge_dev/#cluster-roles).
+
```

Comment:
I don't like this section structure: setting up a git repository is optional in general (if git isn't installed, it will be skipped). 

I suggest this:

This will create a simple Cartridge application that contains:

* one custom role with simple HTTP endpoint;
* tests examples and basic test helpers;
* files required for development (such as `.luacheckrc` etc.).

If you have Git installed, the git repo with the initial commit will be set up and tagged a `0.1.0` version (as a result, `.gitignore` file will be added in the project root).

Reply:
Moreover, `create` command has an option `--template` (it's set to `cartridge` by default, in fact it's the only one template supported). Maybe we should mention that this is the `cartridge` template, not some template?

Reply:
About files structure (I just copied the text from one PR, but I hope it will help):

```lua
-- ├── ${project_name_lower}-scm-1.rockspec  <- rockspec
-- ├── app
-- │   └── roles
-- │       └── custom.lua <- example custom role with simlpe HTTP API
--                                              can be enabled as `app.roles.custom`
-- ├── init.lua <-  file which is the entry point for your application.
--------------- special_files --------------  <- Files used to build and pack application
-- ├── cartridge.pre-build
-- ├── cartridge.post-build
-- ├── Dockerfile.build.cartridge
-- ├── Dockerfile.cartridge
----------------- dev_files ----------------   <- files used for development
-- ├── deps.sh <- IMPORTANT: this file is used to install test deps (such as luatest)
-- ├── instances.yml  <- instances config file (used by cartridge start)
-- ├── .cartridge.yml <- cartridge config file (used by cartridge start)
-- └── tmp <- dir for tmpfiles (used as a run dir, see .cartridge.yml )
--     └── .keep                
----------------- test_files ----------------   <- sample tests
-- ├── test
-- │   ├── helper
-- │   │   ├── integration.lua
-- │   │   └── unit.lua
-- │   ├── helper.lua
-- │   ├── integration
-- │   │   └── api_test.lua
-- │   └── unit
-- │       └── sample_test.lua
---------------- config_files --------------- 
-- ├── .luacheckrc
-- ├── .luacov
-- ├── .editorconfig
--
```

Reply:
> Moreover, `create` command has an option `--template` (it's set to `cartridge` by default, in fact it's the only one template supported). Maybe we should mention that this is the `cartridge` template, not some template?

Let's keep things simple. I was very happy when the unused `simple` template was removed. I suggest saying nothing about the --template option because users don't need it now, and write about it when/if we add more templates.